reframe/core/modules.py

```diff
@@ -681,6 +706,17 @@ def name(self):
     def _module_command_failed(self, completed):
         return completed.stdout.strip() == 'false'
 
+    def available_modules(self):
+        avail = []
+        completed = self._run_module_command(
+            '-t', 'available', msg="could not run module available")
```

Comment:
Why don't you use spider for Lmod?

Reply:
Because spider can be cached. But that's a great idea too.
Do you want me to use it?